src/crafting/FurnaceRecipeManager.php

```diff
@@ -64,6 +65,17 @@ public function register(FurnaceRecipe $recipe) : void{
 		}
 	}
 
+	public function unregister(FurnaceRecipe $recipe) : void{
+		$index = array_search($recipe, $this->furnaceRecipes, true);
+		if($index !== false){
+			unset($this->furnaceRecipes[$index]);
+
+			foreach($this->recipeRegisteredCallbacks as $callback){
+				$callback($recipe);
```

Comment:
This is misleading. Users of these callbacks expect that this will only be called when a recipe is added.

Reply:
This misleading behaviour is worse than breaking BC - it's a silent behavioural change which gives the callee no indication what's actually happening

Reply:
In fact, the creation of new observers is necessary.

Reply:
Yeah, this solution isn't going to cut it.